### ISSUE TITLE
[FEATURE] Enable users upload images

### ISSUE THREAD

**Issue by mathiasayivor** [Dun-sin/Whisper#422]
### Description

It would be great to enable users upload images. The images can be simple, with size limits. They would be removed once the chat is closed to save up space

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

None

**Comment by nishitxmehta** [comment]
Hi @Dun-sin, please assign me this issue, I can do this.

**Comment by Dun-sin** [comment]
> Hi @Dun-sin, please assign me this issue, I can do this.

Read the project guidelines and don't spam else you will be blocked

**Comment by shubham-shinde-442** [comment]
I Would Like to Work on this issue , Can you assign me to get started .

**Comment by Dun-sin** [comment]
> I Would Like to Work on this issue , Can you assign me to get started .

walk me through how you would

**Comment by azr-arch** [comment]
nsfw check is required or just upload image? @Dun-sin

**Comment by Dun-sin** [comment]
>   nsfw check is required or just upload image? @Dun-sin 

Please read the above discussions

**Comment by Dun-sin** [comment]
>   nsfw check is required or just upload image? @Dun-sin 

Hmmm pretty much, would be a great idea but we need to do something with that information. Rather than just not upload the image if it's +18, we allow the users to see it or not

**Comment by drvnprgrmr** [comment]
> > nsfw check is required or just upload image? @Dun-sin
> 
> Hmmm pretty much, would be a great idea but we need to do something with that information. Rather than just not upload the image if it's +18, we allow the users to see it or not

@Dun-sin We could just inform the user and tell them to choose another image.

**Comment by drvnprgrmr** [comment]
Btw, can I work on this?

**Comment by Dun-sin** [comment]
> > > nsfw check is required or just upload image? @Dun-sin
> > 
> > 
> > Hmmm pretty much, would be a great idea but we need to do something with that information. Rather than just not upload the image if it's +18, we allow the users to see it or not
> 
> @Dun-sin We could just inform the user and tell them to choose another image.

good point i agree, do you have a library in mind to use?

**Comment by Dun-sin** [comment]
> Btw, can I work on this?

Definitely, thanks for wanting to contribute, read the issue description properly and understand what it tells you to do also feel free to ask questions, make sure you follow the rules [👉🏽here👈🏽](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md#rules), else your PR will not be accepted and will be closed. Good Luck.

**Comment by drvnprgrmr** [comment]
> > > > nsfw check is required or just upload image? @Dun-sin
> > > 
> > > 
> > > Hmmm pretty much, would be a great idea but we need to do something with that information. Rather than just not upload the image if it's +18, we allow the users to see it or not
> > 
> > 
> > @Dun-sin We could just inform the user and tell them to choose another image.
> 
> good point i agree, do you have a library in mind to use?

Yeah, I was thinking of this: https://github.com/infinitered/nsfwjs

**Comment by Dun-sin** [comment]
@drvnprgrmr 
according to the contributing guideline, your time is up.

**Comment by mathiasayivor** [comment]
@Dun-sin I would like to work on this.

**Comment by Dun-sin** [comment]
will keep this locked, till #558 is merged
